### ISSUE TITLE
Request: Prefix to set Locked layers or Locked layers are Considered foreground or background

### ISSUE THREAD

**Issue by Shadow-Kami** [antipalindrome/Photoshop-Export-Layers-to-Files-Fast#204]
I have used this script extensively because of its versatility, however I have been forced to merge my layers and set them as background or foreground for them to be included in each of my variations of that image as I am now doing factored image instancing.

It would be useful if I could specify what layers can be locked in place and exported into my batch at once that aren't just the background or foreground for pure image modulation.

It would speed up my method of layering and coloring hundreds of graphics at the same time!

**Comment by antipalindrome** [comment]
Let me make sure I understand -- are you asking for a feature where instead of using "Top Layer is Foreground" or "Bottom Layer as Background" that instead you could specify a layer as Foreground or Background?

**Comment by ShadowsAreBlack** [comment]
> Let me make sure I understand -- are you asking for a feature where instead of using "Top Layer is Foreground" or "Bottom Layer as Background" that instead you could specify a layer as Foreground or Background?

Yes, and if you can have it so we can select any layer, I use the script to make hundreds of iterations and with different color outputs via clipping mask, but I have to do them in segments, because I am limited to just the top and bottom layer as the layers saved in all instances.

However I noticed when doing full 200 frame animations having the top and bottom layer as solid makes the script very slow and it almost breaks, so I am unsure if it is viable unless it is optimized well

I am using this script to its full extent daily, basically the PNG that is **locked** in will appear in every export if it can be defined and if it is possible to select multiple would be a game changer for me

My method sort of works like this

Top Layer = Unchanged Accent graphics

Anything in between = colors with blending modes that are clipped to the object below

Bottom Layer = Greyscale Graphic

Export = Graphic * Colors

However this is not the only thing I do with this script

I've also recently discovered the Groups as Layers, which solved a bit of my problems I was having since I just throw all of the bottom excess layers into a group and export it as one but I still have to do a multi process export for other more complicated graphics.


thanks for replying this script has helped me shape how I do graphic design
